openapi-core/src/main/java/io/ballerina/openapi/core/GeneratorUtils.java

```diff
@@ -339,6 +340,27 @@ public static String getValidName(String identifier, boolean isSchema) {
         }
     }
 
+    /**
+     * This util function is for updating the list of nodes {@link TypeDefinitionNode}.
+     * It updates the list while checking the duplicates.
+     *
+     * @param typeName    - Given node name
+     * @param typeDefNode - Generated node
+     * @param typeDefinitionNodeList - Current node list
+     */
+    public static void updateTypeDefNodeList(String typeName, TypeDefinitionNode typeDefNode,
+                                                  List<TypeDefinitionNode> typeDefinitionNodeList) {
+        if (!typeDefinitionNodeList.isEmpty()) {
+            boolean isExit = typeDefinitionNodeList.stream().anyMatch(node ->
+                    (node.typeName().text().trim().equals(typeName)));
+            if (!isExit) {
+                typeDefinitionNodeList.add(typeDefNode);
+            }
+        } else {
+            typeDefinitionNodeList.add(typeDefNode);
+        }
```

Comment:
IMO we don't need these if else clauses since streams can handle empty lists. We can simply use anyMatch and update the list if there are no matches.

Reply:
Yes. Removed the else statement